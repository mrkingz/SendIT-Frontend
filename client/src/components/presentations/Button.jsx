/* eslint-disable react/forbid-prop-types */
import React from "react";
import classnames from "classnames";
import { PropTypes } from "prop-types";

const Button = ({
  wrapperStyle,
  btnStyle,
  onClick,
  isDisabled,
  text,
  children
}) => {
  return (
    <div
      className={classnames({
        [wrapperStyle]: !!wrapperStyle
      })}
    >
      <button
        className={`btn ${btnStyle}`}
        disabled={isDisabled}
        onClick={onClick}
      >
        {children} {text}
      </button>
    </div>
  );
};

Button.propTypes = {
  children: PropTypes.any,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string,
  wrapperStyle: PropTypes.string,
  btnStyle: PropTypes.string
};

export default Button;
