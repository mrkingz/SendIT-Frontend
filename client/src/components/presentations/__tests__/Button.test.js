import React from "react";
import { shallow } from "enzyme";
import Button from "../Button";

const props = {
  message: "Message",
  styles: "style",
  onClick: jest.fn(),
  isDisabled: false,
  text: "",
  btnStyle: "style",
  children: "<h1>Head</h1>"
};

const wrapper = shallow(<Button {...props} />);

describe("Button", () => {
  it("should render Button", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
