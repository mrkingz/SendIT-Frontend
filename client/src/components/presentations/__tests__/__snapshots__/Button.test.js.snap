// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<div
  className=""
>
  <button
    className="btn style"
    disabled={false}
    onClick={[MockFunction]}
  >
    &lt;h1&gt;Head&lt;/h1&gt;
     
  </button>
</div>
`;
